xilem/examples/stopwatch.rs

```diff
@@ -0,0 +1,255 @@
+            .last_start_time
+            .expect("stop should only be called when the start time is set")
+            .elapsed();
+        match dur_since_last_instant {
+            Ok(dur) => {
+                self.added_duration = self.added_duration.add(dur);
+                self.displayed_error = "".into();
+            }
+            Err(err) => {
+                self.displayed_error = format!("failed to calculate elapsed time: {}", err);
+            }
+        }
+        self.last_start_time = None;
+        self.active = false;
+        self.update_display();
+    }
+
+    fn reset(&mut self) {
+        self.active = false;
+        self.last_start_time = None;
+        self.added_duration = Duration::ZERO;
+        self.completed_lap_splits.clear();
+        self.split_start_time = Duration::ZERO;
+        self.update_display();
+    }
+
+    fn lap(&mut self) {
+        let split_end = self.get_current_duration();
+        self.completed_lap_splits
+            .push(split_end.sub(self.split_start_time));
+        self.split_start_time = split_end;
+    }
+
+    fn get_current_duration(&self) -> Duration {
+        match self.last_start_time {
+            Some(last_start_time) => match last_start_time.elapsed() {
+                Ok(elapsed) => self.added_duration + elapsed,
+                Err(err) => {
+                    warn!("error calculating elapsed time: {}", err.to_string());
+                    self.added_duration
+                }
+            },
+            _ => self.added_duration,
+        }
+    }
+
+    fn update_display(&mut self) {
+        self.displayed_duration = self.get_current_duration();
+    }
+}
+
+fn get_formatted_duration(dur: &Duration) -> String {
+    let seconds = dur.as_secs_f64() % 60.0;
+    let minutes = (dur.as_secs() / 60) % 60;
+    let hours = (dur.as_secs() / 60) / 60;
+    format!("{hours}:{minutes:0>2}:{seconds:0>4.1}")
+}
+
+fn app_logic(data: &mut Stopwatch) -> impl WidgetView<Stopwatch> {
+    fork(
+        flex((
+            FlexSpacer::Fixed(5.0),
+            label(get_formatted_duration(&data.displayed_duration)).text_size(70.0),
+            flex((lap_reset_button(data), start_stop_button(data))).direction(Axis::Horizontal),
+            FlexSpacer::Fixed(1.0),
+            laps_section(data),
+            label(data.displayed_error.as_ref()),
+        )),
+        data.active.then(|| {
+            // Only update while active.
+            async_repeat(
+                |proxy| async move {
+                    let mut interval = time::interval(Duration::from_millis(50));
+                    loop {
+                        interval.tick().await;
+                        let Ok(()) = proxy.message(()) else {
+                            break;
+                        };
+                    }
+                },
+                |data: &mut Stopwatch, ()| {
+                    data.update_display();
+                },
+            )
```

Comment:
I mean totally unrelated to this PR, just some food for thought, as I'm currently thinking about something like a `request_animation_frame` view in xilem_web.
(or even let xilem_webs event loop run inside something like `request_animation_frame` with batched events/mutations, but I need to check how that impacts input latency).

I think it might be interesting to have a view that is synced with the framerate/vsync/compositor, to e.g. have a smooth increasing counter (or more useful-cases such as a canvas animating/moving stuff).

Reply:
That would definitely be ideal for this example.